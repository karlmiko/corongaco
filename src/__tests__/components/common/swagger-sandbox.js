/* eslint-disable no-global-assign */

import React from 'react'
import renderer from 'react-test-renderer'
import SwaggerUI from 'swagger-ui'
import SwaggerSandbox from '../../../components/common/swagger-sandbox'

beforeEach(() => {
  window.SwaggerUI = SwaggerUI
})

describe('Components : Common: Swagger sandbox', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SwaggerSandbox />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Common: Swagger sandbox : Undefined window', () => {
  it('renders correctly', () => {
    window = undefined
    const tree = renderer.create(<SwaggerSandbox />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

afterEach(() => {
  delete window.SwaggerUI
})
