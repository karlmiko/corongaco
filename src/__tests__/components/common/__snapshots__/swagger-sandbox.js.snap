// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Common: Swagger sandbox : Undefined window renders correctly 1`] = `
<div
  id="swaggerWrapper"
/>
`;

exports[`Components : Common: Swagger sandbox renders correctly 1`] = `
<div
  id="swaggerWrapper"
/>
`;
